Alberto-todo-list/src/components/Container.jsx

```diff
@@ -1,40 +1,57 @@
-/* Declaracion de dominios*/ 
-import React, { useState } from "react";
-
+import { useState } from "react";
 
 /*Importar los componentes que se van a utilizar en el componente actual
-En este caso se utilizan los componentes principales del Check list para 
+En este caso se utilizan los componentes principales del Check list para
 organizarlos en el container principal
-*/ 
+*/
 import TaskList from "./TaskList";
 import FormTodo from "./FormTodo";
 
-const Container = () => {
-
-  /* Se crea una variable de estado (Hook) en donde iremos guardando todas las actividades 
+const TodoList = () => {
+  /* Se crea una variable de estado (Hook) en donde iremos guardando todas las actividades
   que el usuario desea realizar, al utilizar estados en reac se generan dos variables
     list -> listado de actividades
     setList -> Para actializar el estado actual
   */
-  const [list, setList] = useState([]); 
-  
- /* handleAddItem Funcion para actualizar los valores en la lista
+  const [list, setList] = useState([]);
+
+  /* handleAddItem Funcion para actualizar los valores en la lista
     considerando las acciones que se ejecutan en FormTodo
     Los valores se le dan como estados de entrada a el TaskList
  */
-  
-  const handleAddItem = addItem => {
-    setList([...list, addItem]); 
+
+  const handleAddItem = (addItem) => {
+    setList([...list, addItem]);
+  };
+
+  /** Cambiar a realizado el es estado de la actividad cuando se chequee*/
+  const onChangeStatus = (item) => {
+    const updatedList = list.map((listItem) => {
+      if (listItem.id === item.id) {
+        return { ...listItem, done: !listItem.done };
+      }
+      return listItem;
+    });
+    setList(updatedList);
+  };
+
+  /** remover todos los items chequeados con el filtro */
+  const onClickRemoveItem = () => {
+    const updateList = list.filter((item) => !item.done);
+    setList(updateList);
   };
 
   return (
     <div>
-      
       <FormTodo handleAddItem={handleAddItem} />
-      
-      <TaskList list={list} setList={setList} />
+
+      <TaskList
+        list={list}
+        onChangeItem={onChangeStatus}
+        onCleanUpCompletedTasks={onClickRemoveItem}
+      />
```

Comment:
about this change, this is my recommendation, instead of passing the whole `setList()` function, use the "handlers" that are defined in the parent. 

Since this is exactly the same solution from one of your colleges (im not mad), im going to repost the video i did for another correction.

Reply:
https://github.com/user-attachments/assets/eee1b8fc-935a-4c81-b6f6-438c6c13db42